pkg/collector/check/check.go

```diff
@@ -213,6 +214,51 @@ func (c *Config) IsTemplate() bool {
 	return false
 }
 
+// IsJMX returns if the config is for a JMX check
+func (c *Config) IsJMX() bool {
+	return IsConfigJMX(c.String(), c.InitConfig)
+}
+
+// AddMetrics adds metrics to a check configuration
+func (c *Config) AddMetrics(metrics ConfigData) error {
+	var rawInitConfig map[interface{}]interface{}
+	err := yaml.Unmarshal(c.InitConfig, &rawInitConfig)
+	if err != nil {
+		return err
+	}
+
+	var rawMetricsConfig []interface{}
+	err = yaml.Unmarshal(metrics, &rawMetricsConfig)
+	if err != nil {
+		return err
+	}
+
+	// JMX fetch expects the metrics to be a part of init_config, under "conf"
+	// Preserve any metrics currently in init_config
```

Comment:
Is there a way to completely override the `.metrics` file from the template? Looks like we unconditionality merge them, but I think having the option of disabling the common metrics might be requested down the road.

Reply:
Agreed, we should add something like `collect_default_metrics: false` in the `init_config` section of the config file.

Reply:
Good idea 👍